tests_e2e/tests/ext_policy/policy_dependencies_cases.py

```diff
@@ -0,0 +1,243 @@
+    return policy, template, expected_errors, deletion_order
+
+
+def _should_fail_multi_config_depends_on_disallowed_no_config():
+    template = [
+        __get_extension_template(VmExtensionIds.AzureMonitorLinuxAgent),
+        __get_extension_template(VmExtensionIds.RunCommandHandler, depends_on=["AzureMonitorLinuxAgent"])
+    ]
+    policy = \
+        {
+            "policyVersion": "0.1.0",
+            "extensionPolicies": {
+                "allowListedExtensionsOnly": True,
+                "extensions": {
+                    "Microsoft.CPlat.Core.RunCommandHandlerLinux": {}
+                }
+            }
+        }
+    expected_errors = [
+        "[ExtensionPolicyError] Extension will not be processed: failed to enable extension 'Microsoft.Azure.Monitor.AzureMonitorLinuxAgent' because extension is not specified in allowlist",
+        "VM has reported a failure when processing extension 'RunCommandHandlerLinux' (publisher 'Microsoft.CPlat.Core' and type 'RunCommandHandlerLinux'). Error message: 'Skipping processing of extensions since execution of dependent extension Microsoft.Azure.Monitor.AzureMonitorLinuxAgent failed'."
+    ]
+    deletion_order = [VmExtensionIds.RunCommandHandler, VmExtensionIds.AzureMonitorLinuxAgent]
+    return policy, template, expected_errors, deletion_order
+
+
+def _should_succeed_single_config_depends_on_no_config():
+    template = [
+        __get_extension_template(VmExtensionIds.AzureMonitorLinuxAgent),
+        __get_extension_template(VmExtensionIds.CustomScript, depends_on=["AzureMonitorLinuxAgent"])
+
+    ]
+    policy = \
+        {
+            "policyVersion": "0.1.0",
+            "extensionPolicies": {
+                "allowListedExtensionsOnly": True,
+                "extensions": {
+                    "Microsoft.Azure.Monitor.AzureMonitorLinuxAgent": {},
+                    "Microsoft.Azure.Extensions.CustomScript": {}
+                }
+            }
+        }
+    expected_errors = []
+    deletion_order = [VmExtensionIds.CustomScript, VmExtensionIds.AzureMonitorLinuxAgent]
+    return policy, template, expected_errors, deletion_order
+
+
+def _should_succeed_single_config_depends_on_single_config():
+    template = [
+        __get_extension_template(VmExtensionIds.CustomScript),
+        __get_extension_template(VmExtensionIds.VmAccess, depends_on=["CustomScript"])
+    ]
+    policy = \
+        {
+            "policyVersion": "0.1.0",
+            "extensionPolicies": {
+                "allowListedExtensionsOnly": True,
+                "extensions": {
+                    "Microsoft.Azure.Extensions.CustomScript": {},
+                    "Microsoft.OSTCExtensions.VMAccessForLinux": {}
+                }
+            }
+        }
+    expected_errors = []
+    deletion_order = [VmExtensionIds.VmAccess, VmExtensionIds.CustomScript]
+    return policy, template, expected_errors, deletion_order
+
+
+def _should_no_dependencies():
```

Comment:
Looks like this may be leftover code, I don't see it referenced

Reply:
Removed, thanks!